diff --git a/webapp/graphite/util.py b/webapp/graphite/util.py
index 89709d8..914d1e4 100644
--- a/webapp/graphite/util.py
+++ b/webapp/graphite/util.py
@@ -12,6 +12,7 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License."""
 
+import sys
 try:
   import cPickle as pickle
   USING_CPICKLE = True
@@ -88,7 +89,10 @@ if USING_CPICKLE:
   class SafeUnpickler(object):
     PICKLE_SAFE = {
       'copy_reg': set(['_reconstructor']),
-      '__builtin__': set(['object']),
+      '__builtin__': set(['object', 'list']),
+      'collections': set(['deque']),
+      'graphite.render.datalib': set(['TimeSeries']),
+      'graphite.intervals': set(['Interval', 'IntervalSet']),
     }
 
     @classmethod
@@ -111,7 +115,10 @@ else:
   class SafeUnpickler(pickle.Unpickler):
     PICKLE_SAFE = {
       'copy_reg': set(['_reconstructor']),
-      '__builtin__': set(['object']),
+      '__builtin__': set(['object', 'list']),
+      'collections': set(['deque']),
+      'graphite.render.datalib': set(['TimeSeries']),
+      'graphite.intervals': set(['Interval', 'IntervalSet']),
     }
 
     def find_class(self, module, name):
